app/scripts/views/settings/communication_preferences.js

```diff
@@ -35,11 +41,15 @@ function (Cocktail, Xss, Constants, MarketingEmailErrors, Metrics, BaseView, For
       return self._marketingEmailPrefs;
     },
 
-    beforeRender: function () {
+    afterVisible: function () {
       var self = this;
       var emailPrefs = self.getMarketingEmailPrefs();
+
       return emailPrefs.fetch()
-        .fail(function (err) {
+        .then(function () {
+          return self.render();
```

Comment:
For this PR this approach is fine, but it does allow for a flash of button content if the user reloads the page with the communications_preferences panel open:

![subscribe-unsubscribe-flash](https://cloud.githubusercontent.com/assets/848085/10164207/23c1d416-66b0-11e5-8ca1-d81aa5c312b2.gif)

A solution to this is to render the page with a "loading" button (gray button with throbber) while the fetch is outstanding, then re-render the button (or the entire screen) with the appropriate text once the emailPrefs fetch has completed.

Since re-loading the page with the communication prefs screen is probably not a common occurrence, "meh" for now.